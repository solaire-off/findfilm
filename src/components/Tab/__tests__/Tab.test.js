import React from "react";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Tab } from "..";

const TabCaption = "All";

it("Current render", () => {
  const { asFragment } = render(
    <Tab
      key={TabCaption}
      name={TabCaption}
      isActive={true}
      onClick={() => {}}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});

it("Callback onClick work", () => {
  const mockCallback = jest.fn();
  render(
    <Tab
      key={TabCaption}
      name={TabCaption}
      isActive={false}
      onClick={mockCallback}
    />
  );

  userEvent.click(screen.getByText(TabCaption));
  expect(mockCallback).toBeCalledTimes(1);
});
