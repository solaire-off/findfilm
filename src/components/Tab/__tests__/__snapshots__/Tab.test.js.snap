// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render 1`] = `
<DocumentFragment>
  <button
    class="tabs-list__item tabs-list__item--active"
    type="button"
  >
    All
  </button>
</DocumentFragment>
`;
